### ISSUE TITLE
In-Task Dependent Task Informational Alert

### ISSUE THREAD

**Issue by carye3** [newjersey/business.nj.gov#468]

**Comment by carye3** [comment]
@Jesslax flagging that we need to identify which tasks in roadmap are parent / child to carry out this ticket

**Comment by aloverso** [comment]
@Jesslax we don't need *all* of the relationships defined in order to complete this story. just one or two. but if you were to show us how you're keeping track of that, it would help a lot in terms of how we build out the relationship definitions

**Comment by Jesslax** [comment]
@aloverso The first sentence of a task screen identifies dependencies for other task screens.

Getting the EIN is dependent on the business name
Forming your business is dependent on the business name, and having your EIN

**Comment by Jesslax** [comment]
@aloverso @carye3 The way we are currently showing this kind of dependency--a dependency on step 3--is by saying there is a dependency on form your business entity and register for state taxes tasks.

**Comment by carye3** [comment]
@aloverso agreed with Jess -- this shouldn't change design, dependencies still just exist among tasks, not sections

**Comment by skeeJay** [comment]
I don't think it makes sense to QA this until it can be done in combination with #470, so I'm going to wait on that.

**Comment by skeeJay** [comment]
@Jesslax Would be good to get your eyes on this, see if it's behaving as desired. Also, I'm curious whether the text we landed on is too limiting… do we always want to say that you need to "complete" prior tasks, or should we suggest something a little looser like: "Before starting this task, we recommend that you take a look at these: Form Your Business Entity, Register for State Taxes"

**Comment by Jesslax** [comment]
@skeeJay @aloverso Can we have default copy and unique copy? Or as it's currently built is there only default copy? I will take a look at the copy.

**Comment by aloverso** [comment]
@Jesslax @skeeJay copy is stored here https://github.com/newjersey/business.nj.gov/blob/main/web/display-content/tasks/TaskDefaults.ts

Can you describe more what you mean by default and unique copy?

**Comment by skeeJay** [comment]
@aloverso I think Jess is asking if the copy can be overridden/customized on a per-Task basis. I suspect you did not implement that, because we didn't ask for it.

**Comment by Jesslax** [comment]
@skeeJay That works. I am imaging that this alert banner replaces the copy below it, like "After you found an available business name, you can form your business." Is that what you were thinking too? 
There are a few more task screens that need to have the dependencies alert banner.

**Comment by skeeJay** [comment]
@Jesslax Yes, once we have this feature, you can certainly update any Task text to remove redundancy that we had previously just been relying on. What are the other Tasks that are missing dependencies? Is that the kind of thing you are meant to be able to edit yourself with this feature?

**Comment by Jesslax** [comment]
I believe I should be able to add this to any task I want, right @aloverso

**Comment by skeeJay** [comment]
I believe so. Okay, I think we have some QA edits, which I have added to the description. Adding the Needs Work label.

**Comment by aloverso** [comment]
@skeeJay thanks for the catch, I did miss the piece about removing it from the alert. working on it now

For the QA note about wording, is it ok if I push that one back to you & Jess? I think it's a good idea to get in the practice of letting you all change that whenever you like without it being part of the story acceptance. 

[Here's the file to edit](https://github.com/newjersey/business.nj.gov/blob/main/web/display-content/tasks/TaskDefaults.ts) let me know what you think of that idea

**Comment by skeeJay** [comment]
@aloverso Yup, sounds good about the copy.

**Comment by aloverso** [comment]
@Jesslax yes, you can add it to any task. [Here's the file for storing dependencies](https://github.com/newjersey/business.nj.gov/blob/main/web/roadmaps/task-dependencies.json)

If it's confusing, I am happy to walk it through with you on a call, but here's the gist

Each line looks like, for example:
```
  "form-business-entity": ["search-business-name", "register-for-ein"],
```

This means that form-business-entity is *blocked by* the two tasks on the right. They're in square brackets `[]` because that indicates a list. You can have a list even of one item like this

``` 
 "register-for-ein": ["search-business-name"],
```

And adding to this structure will automatically do the reverse relationship when the other story gets delivered. So you can add the line above, and it will make sure that BOTH register-for-ein shows you should complete seach-business-name first, AND that search-business-name informs you that it unlocks register-for-ein

let me know if you have questions!

**Comment by skeeJay** [comment]
@Jesslax You should definitely look at this one from a QA perspective, since this is the "good" one.

**Comment by aaronvux** [comment]
Sorry.  I was confused by the comment.  I'll move it back.
